libraries/AP_Camera/AP_RunCam.cpp

```diff
@@ -1058,8 +1069,86 @@ uint8_t AP_RunCam::Request::get_expected_response_length(const Command command)
     return 0;
 }
 
+// AP_Camera API
+
+// return true if healthy
+bool AP_RunCam::healthy() const
+{
+    return camera_ready();
+}
+
+// momentary switch to change camera between picture and video modes
+void AP_RunCam::cam_mode_toggle() {
+
+}
+
+// entry point to actually take a picture.  returns true on success
+bool AP_RunCam::trigger_pic() {
+    return false;
+}
+
+// send camera information message to GCS
+void AP_RunCam::send_camera_information(mavlink_channel_t chan) const
+{
+    // exit immediately if not initialised
+    if (!camera_ready()) {
+        return;
+    }
+
+    static const uint8_t vendor_name[32] = "RunCam";
+    uint8_t model_name[32] {};
+    strncpy((char *)model_name, _models[_cam_type.get()-1], MIN(sizeof(model_name), sizeof(_models[_cam_type.get()-1])));
```

Comment:
This looks very bad - indexing an array from a user-set parameter looks like a really bad idea.

What happens if you change CAM1_TYPE to 0 after it has detected a runcam?

Reply:
Maybe check the configured type is the same as the allocated type in here?

Reply:
No, no - check the if at the top of the function

Reply:
Looks right now

Reply:
OK fixed